composer/loggers/mosaicml_logger.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+import logging
+import os
+import sys
+import warnings
+from queue import Queue
+from threading import Thread
+from typing import Optional
+
+import requests
+
+from composer.core.logging import LogLevel, RankZeroLoggerBackend, TLogData
+from composer.core.types import JSON, Logger, State, StateDict
+
+_MOSAICML_API_KEY_ENV = "MOSAICML_LOGGER_API_KEY"
+_MOSAICML_LOGGER_URL = "https://api.mosaicml.com/v0/log/metric"
+_JOB_ID_ENV = "MOSAICML_LOGGER_JOB_ID"
+
+_STOP_LOGGING_SIGNAL = "STOP_LOGGING"
+
+log = logging.getLogger(__name__)
+
+
+def _send_data(job_id: str, data: JSON):
+    try:
+        response = requests.post(_MOSAICML_LOGGER_URL,
+                                 headers={"X-MosaicML-API-key": os.environ.get(_MOSAICML_API_KEY_ENV, "")},
+                                 json={
+                                     "experimentID": job_id,
+                                     "data": data,
+                                 },
+                                 timeout=5)
+        if response.status_code >= 400:
+            # Terminate the process in the case of a non-transient error
+            log.error("Posting data to MosaicML backend failed with response code "
+                      f"{response.status_code} and message {response.text}.")
+            sys.exit(1)
```

Comment:
You could consider just to drop the batch, instead of killing the training job

Reply:
Discussed this with @ravi-mosaicml - we drop the batch in the case of a transient error but thought it was best to just terminate the process in the case of a non-transient error because this usually means that none of the batches will be logged (ex. the user provided an incorrect api key or something).